### ISSUE TITLE
[Bug][2.5xmr reward for fixing] gui is not clearing screen on Whonix OS

### ISSUE THREAD

**Issue by 4x00** [monero-project/monero-gui#4187]
Hello

On Whonix OS there is serious issue that when clicking or navigating or doing any interaction that renders to window, the previous content is not cleared from screen and new content is drawn on top of it without clearing previous content and it becomes a mess, the workaround that I am using is to move the window after every of my interaction with the window - moving the window re-renders the gui and makes it usable until my next interaction
I would be very grateful if someone looked into this issue and would fix it and I can offer 2.5xmr to whoever fixes it... please contact me on 4x0@tuta.io for your reward

Sincerely

**Comment by selsta** [comment]
Can you make a screenshot? Also which binaries are you using?

**Comment by 4x00** [comment]
Newest binaries fron getmonero.org-- 
 Sent with Tutanota, enjoy secure & ad-free emails. 



Jun 7, 2023, 17:28 by ***@***.***:

**Comment by 4x00** [comment]
I also noticed this bug only affects transactions and receive tab

steps to repro
) click on transactions tab -> window does not update until I move/resize window
) click on tranasction while being on transactions tab -> window does not update until I move/resize window
) click on receive tab -> window does not update until I move/resize window

**Comment by 4x00** [comment]
) also if I move away from receive or transactions tab -> window does update until I move/resize window
also hovering over elements that drawing was skipped for will draw them

I do not want to show my xmr transactions/addresses and posting screenshots would exactly that

**Comment by selsta** [comment]
Can you reproduce the issue when starting with the following env var?

```
QMLSCENE_DEVICE=softwarecontext ./monero-wallet-gui
```

**Comment by plowsof** [comment]
skip to 41 seconds please. is this the issue you encounter? (i am using whonix in virtualbox with no modifications)
[Whonix-Workstation-XFCE-screen0.webm](https://github.com/monero-project/monero-gui/assets/77655812/4ba34187-8040-403c-8223-13255ada1807)

**Comment by 4x00** [comment]
yep looks like the same, I did not try if it happens in wallet creation process though

**Comment by 4x00** [comment]
with large transaction histories and 20+ recieve addresses it becomes quite annoying (but not impossible) to use monero-wallet-gui

**Comment by codingdandy** [comment]
Give it a try: 

* `QT_OPENGL=software ./monero-wallet-gui`
* `QT_XCB_FORCE_SOFTWARE_OPENGL=1 ./monero-wallet-gui`
* `LIBGL_ALWAYS_SOFTWARE=1 ./monero-wallet-gui`
* `QT_OPENGL=angle ./monero-wallet-gui`
* `QT_OPENGL=desktop ./monero-wallet-gui`

**Comment by 4x00** [comment]
I tested with all of the parameters on the newest monero-wallet-gui and found no difference, sadly
on a bright note - I learned that when I try to screenshot the gui with alt+prtsc the gui corrects itself and renders the correct result

**Comment by 4x00** [comment]
> From what I'm seeing here you're using Whonix inside a virtual machine right? You aren't on Qubes OS or anything like that?

Yes my setup involves Whonix in virtualbox with host os being ubuntu 22.04

**Comment by selsta** [comment]
I worry that this is a Whonix bug or Qt bug. We don't have any custom render code and use the latest version of Qt 5.

**Comment by 4x00** [comment]
> > > From what I'm seeing here you're using Whonix inside a virtual machine right? You aren't on Qubes OS or anything like that?
> > 
> > 
> > Yes my setup involves Whonix in virtualbox with host os being ubuntu 22.04
> 
> I've tried to reproduce the bug in virtualbox on Windows 11 but everything was smooth. I suspect it might have something to do with the fact you're running it on Ubuntu and there might be some drivers issues or something related that is responsible for the glitching.
> 
> Are you running the latest Whonix release?

I am running virtualbox 6.1.38_Ubuntu r153438

**Comment by 4x00** [comment]
> > > > > From what I'm seeing here you're using Whonix inside a virtual machine right? You aren't on Qubes OS or anything like that?
> > > > 
> > > > 
> > > > Yes my setup involves Whonix in virtualbox with host os being ubuntu 22.04
> > > 
> > > 
> > > I've tried to reproduce the bug in virtualbox on Windows 11 but everything was smooth. I suspect it might have something to do with the fact you're running it on Ubuntu and there might be some drivers issues or something related that is responsible for the glitching.
> > > Are you running the latest Whonix release?
> > 
> > 
> > I am running virtualbox 6.1.38_Ubuntu r153438
> 
> Can you reproduce the bug on the latest release of virtualbox? If you can reproduce it, I'll try to reach out to the Whonix devs.

Sorry for long reply, I didn't want to update virtualbox
now I have updated virtualbox to 7 and can confirm it is still reproducable
